File: Src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
# import ros 
import rclpy 
from rclpy.node import Node ## import  node class

class MyNode(Node):

    def __init__(self):
        super().__init__("py_test")
        self.counter_= 0
        self.get_logger().info("Hello ROS2")
        self.create_timer(0.5, self.timer_callback)

    def timer_callback(self):
        self.counter_ += 1
        self.get_logger().info("Hello" + str(self.counter_))
def main(args=None):
    rclpy.init(args=args) ## initialize ros to create ros communiation 
    node = MyNode()
    rclpy.spin(node) # spin allow to node
    rclpy.shutdown() # ros communication shutdown
    
if __name__== "__main__":
    main()

File: src/my_cp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

int main(int argc,char **argv)
{
    rclcpp::init(argc, argv); // initialize ros2 communication
    auto node = std::make_shared<rclcpp::Node>("cpp_test");
    RCLCPP_INFO(node->get_logger(), "Hello CPP Node");
    rclcpp::spin(node);
    rclcpp::shutdown(); // shutdown 
   return 0;
}